pkg/apis/core/v1alpha1/tiltrun_types.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2020 The Tilt Dev Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource"
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource/resourcestrategy"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TiltRun provides introspective data about the status of the Tilt process.
+// +k8s:openapi-gen=true
+type TiltRun struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   TiltRunSpec   `json:"spec,omitempty"`
+	Status TiltRunStatus `json:"status,omitempty"`
+}
+
+// TiltRunList is a list of TiltRun objects.
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type TiltRunList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []TiltRun `json:"items"`
+}
+
+// TiltRunSpec defines the desired state of TiltRun
+type TiltRunSpec struct {
+	// TiltfilePath is the path to the Tiltfile for the run. It cannot be empty.
+	TiltfilePath string `json:"tiltfilePath"`
```

Comment:
I could see this being optional in the future so that you could theoretically not have a `Tiltfile` at all and purely interact with API -> create `TiltRun` and then (via API) create resources attached to that.

Reply:
Speaking to some of the high-level questions, I think it's reasonable to assume:
- The apiserver will be a persistent process that has a separate lifecycle than the tilt up or tilt ci process
- The Tiltfile is one of many ways to load configuration into that apiserver

And there might be extensions that load configuration differently (like the `updog` command currently)